pcdsdevices/ipm.py

```diff
@@ -102,6 +105,46 @@ class IPMMotion(BaseInterface, Device):
     tab_whitelist = ['target', 'diode', 'insert', 'remove', 'inserted',
                      'removed']
 
+    def format_status_info(self, status_info):
+        """
+        Override status info handler to render the ipm.
+
+        Display ipm status info in the ipython terminal.
+
+        Parameters
+        ----------
+        status_info: dict
+            Nested dictionary. Each level has keys name, kind, and is_device.
+            If is_device is True, subdevice dictionaries may follow. Otherwise,
+            the only other key in the dictionary will be value.
+        Returns
+        -------
+        status: str
+            Formatted string with all relevant status information.
+        """
+        name = ' '.join(self.prefix.split(':'))
+
+        x_motor_pos = status_info.get('diode', {}).get('x_motor', {}).get(
+                                      'position', 'N/A')
+        y_motor_pos = status_info.get('diode', {}).get('state', {}).get(
+                                      'motor', {}).get('position', 'N/A')
+        d_units = status_info.get('diode', {}).get('x_motor', {}).get(
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_pos = status_info.get('target', {}).get('motor', {}).get(
+                                     'position', 'N/A')
+        t_units = status_info.get('target', {}).get('motor', {}).get(
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_state_num = status_info.get('target', {}).get('state', {}).get(
+                                           'value', 'N/A')
+        target_state = status_info.get('target', {}).get('position', 'N/A')
+
+        return f"""\
+{name}: Target {target_state_num} {target_state}
+Target Position: {target_pos} [{t_units}]
+Wave 8 Diode Position(x, y): \
+{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
```

Comment:
Not all IPMs use Wave8s. Can we add something like this?
```suggestion
        if 'ipimb' in status_info.keys():
            diode_type = 'IPIMB '
        elif 'wave8' in status_info.keys():
            diode_type = 'Wave8 '
        else:
            diode_type = ''

        return f"""\
{name}: Target {target_state_num} {target_state}
Target Position: {target_pos} [{t_units}]
{diode_type}Diode Position(x, y): \
{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
```

Reply:
Thanks for pointing this out @ZryletTC !